src/languages/vbnet.js

```diff
@@ -1,73 +1,160 @@
-/*
-Language: Visual Basic .NET
-Description: Visual Basic .NET (VB.NET) is a multi-paradigm, object-oriented programming language, implemented on the .NET Framework.
-Author: Poren Chiang <ren.chiang@gmail.com>
-Website: https://docs.microsoft.com/en-us/dotnet/visual-basic/getting-started/
-*/
+/**
+ * Language: Visual Basic .NET
+ * Description: Visual Basic .NET (VB.NET) is a multi-paradigm, object-oriented programming language, implemented on the .NET Framework.
+ * Authors: Poren Chiang <ren.chiang@gmail.com>, Jan Pilzer
+ * Website: https://docs.microsoft.com/dotnet/visual-basic/getting-started/
+ */
 
 export default function(hljs) {
+  /**
+   * Character Literal
+   * Either a single character ("a"C) or an escaped double quote (""""C).
+   */
+  const CHARACTER = {
+    className: 'string',
+    begin: /"(""|[^/n])"C\b/
+  };
+
+  const STRING = {
+    className: 'string',
+    begin: /"/,
+    end: /"/,
+    illegal: /\n/,
+    contains: [
+      {
+        // double quote escape
+        begin: /""/
+      }
+    ]
+  };
+
+  /**
+   * Date Literals consist of a date, a time, or both separated by whitespace, surrounded by #.
+   */
+  const DATE = {
+    className: 'literal',
+    variants: [
+      {
+        // Date: #M/D/YYYY# or #YYYY-MM-DD#
+        begin: /# *((\d+[-/]){2}\d+) *#/
+      },
+      {
+        // Time: #H:mm[:ss]# (24h) or #h[:mm[:ss]] A# (12h )
+        begin: /# *((\d+(:\d+){0,2} *(AM|PM)?)) *#/
+      },
+      {
+        // Date and time
+        begin: /# *((\d+[-/]){2}\d+ +((\d+(:\d+){0,2} *(AM|PM)?))) *#/
+      }
+    ]
+  };
+
+  const NUMBER = {
+    className: 'number',
+    relevance: 0,
+    variants: [
+      {
+        // Float
+        begin: /-?\b\d[\d_]*((\.[\d_]+(E[+-]?[\d_]+)?)|(E[+-]?[\d_]+))[RFD@!#]?/
+      },
+      {
+        // Integer (base 10)
+        begin: /-?\b\d[\d_]*((U?[SIL])|[%&])?/
+      },
+      {
+        // Integer (base 16)
+        begin: /&H[\dA-F_]+((U?[SIL])|[%&])?/
+      },
+      {
+        // Integer (base 8)
+        begin: /&O[0-7_]+((U?[SIL])|[%&])?/
+      },
+      {
+        // Integer (base 2)
+        begin: /&B[01_]+((U?[SIL])|[%&])?/
+      }
+    ]
+  };
+
+  const LABEL = {
+    className: 'label',
+    begin: /^\w+:/
+  };
+
+  const COMMENT = hljs.COMMENT(null, /$/, {
+    variants: [
+      {
+        begin: /'(?!'')/
+      },
+      {
+        begin: /\bREM\b/
```

Comment:
Pretty sure REM needs a space after it, no?  Many things other than spaces would count as a word boundary but I think you need a space for it to be a comment? Or am I mistaken?

Probably an issue with the \b at the beginning to.

- A space is required between the REM keyword and comment.
- You can put a REM statement alone on a line, or you can put it on a line following another statement. The REM statement must be the last statement on the line. If it follows another statement, the REM must be separated from that statement by a space.

I think this might be a case for now where we need `\sREM ` even though it's annoying.

Reply:
Absolutely correct.
Changed to `([\t ]|^)REM(?!\S)`